### DISCUSSION TITLE
How to configure env variables  for development, stagging and production in Azure App Service for NextJS?

### DISCUSSION THREAD

**Discussion by alamenai** [vercel/next.js#59250] [Help]
### Summary

We deployed our NextJS application in Azure with App Service.

Currently, we have two env files:

**.env.production**
```
# Keys
NEXT_PUBLIC_API_KEY= our_key

# URLs
NEXT_PUBLIC_APIS_BASE_URL=our_base_url
```

**.env.development**

```
# Keys
NEXT_PUBLIC_API_KEY= our_key

# URLs
NEXT_PUBLIC_APIS_BASE_URL=our_base_url
```

I want to configure these variables like [vercel](https://vercel.com/docs/projects/environment-variables?utm_medium=docs&utm_source=next-site&utm_campaign=next-website#preview-environment-variables) in Azure App Service and just PULL THEM from azure for different environments :

* Development
* Staging
* Production

How can I do this?

Thank you

**Comment by ramiborni** [comment]
Did you figure out how to do this? because I'm having same issue now.

**Reply by alamenai** [reply]
Not yet, I'll share if I find a solution. Currently, we put the variables in the env files.

**Comment by theklr** [comment]
If you're using Azure devOps we've found the [secure file](https://learn.microsoft.com/en-us/azure/devops/pipelines/library/secure-files?view=azure-devops) to be the smoothest pattern as the variables both in github and azure aren't as accessible for next
